libpod/image/image.go

```diff
@@ -1076,3 +1078,144 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
 	}
 	return ociv1Img.History[0].Comment, nil
 }
+
+type TreeImage struct {
+	ID       string
+	ParentID string
+	RepoTags []string
+	Size     int64
+}
+
+// Tree gets dependency list in tree format of the image
+func (i *Image) Tree(ctx context.Context, imageName string) (*bytes.Buffer, error) {
+
+	layerMap, err := i.layerToImageMap(ctx)
+	if err != nil {
+		return nil, err
+	}
+	layerMap, err = i.addImageToLayerInfo(ctx, layerMap)
+
+	repoTag, err := i.MatchRepoTag(imageName)
+	if err != nil {
+		return nil, err
+	}
+	// Get the image top layer and then parse as per parent
+	layer := i.TopLayer()
+	dependencyList := returnDependencyTree(layerMap, repoTag, layer)
+
+	var buffer bytes.Buffer
+	printTree(&buffer, dependencyList)
+
+	return &buffer, err
+}
+
+func printTree(buffer *bytes.Buffer, treeStack []TreeImage) {
+
+	prefixSpace := 0
+	for i := len(treeStack) - 1; i >= 0; i-- {
+		for p := 0; p < prefixSpace; p++ {
+			buffer.WriteString(fmt.Sprintf(" "))
+		}
+		buffer.WriteString(fmt.Sprintf("└─ ID: %s  Tags: %s VirtualSize: %v\n",
+			treeStack[i].ID[:12],
+			treeStack[i].RepoTags,
+			units.HumanSizeWithPrecision(float64(treeStack[i].Size), 4)),
+		)
+
+		prefixSpace++
+	}
+}
+
+func (i *Image) layerToImageMap(ctx context.Context) (map[string]*TreeImage, error) {
```

Comment:
This is not making an “image map” ”from” a ”layer”.  A more accurate name would be better.  (The `TreeImage` name for a struct describing a layer, which may have no direct correspondence to any particular image, should also be changed.)

Reply:
The function name is now much better.

Using `LayerMap` for an _element_ of a map is still a bit confusing… but I can’t think of a short enough improvement.